libtock/tock.h

```diff
@@ -127,6 +127,18 @@ typedef struct {
   uint32_t data;
 } memop_return_t;
 
+// Return structure for a Yield-WaitFor syscall. The return value are the
+// arguments that would have been passed to the upcall the the Yield-WaitFor was
+// waiting on.
+typedef struct {
+  // Upcall argument 1.
+  uint32_t data0;
+  // Upcall argument 2.
+  uint32_t data1;
+  // Upcall argument 3.
+  uint32_t data2;
```

Comment:
One nit question: the arguments are `uint32_t`, but above in `libtock/tock.h`, upcall function pointers are defined as `typedef void (subscribe_upcall)(int, int, int, void*)`. Is the difference meaningful? Do we care about it?

To me, it feels like maybe the `subscribe_upcall` signature is what's wrong.

Reply:
Agree that subscribe upcall is older and was created before we got more explicit about syscall types.